### PR TITLE
add list of ids to contingency info export endpoint

### PR DESCRIPTION
**Pull request by jamal-khey**
depend on [security anaysis PR](https://github.com/gridsuite/security-analysis-server/pull/135)

### CODE REVIEW

File: src/main/java/org/gridsuite/actions/server/ContingencyListController.java [hunk 1]
```diff
@@ -111,6 +111,16 @@ public ResponseEntity<List<ContingencyInfos>> exportContingencyInfosList(@PathVa
         return ResponseEntity.ok().body(service.exportContingencyInfosList(id, networkUuid, variantId));
     }
 
+    @GetMapping(value = "/contingency-lists/contingency-infos/export", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Evaluate and export a contingency infos list to JSON format")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The contingency list in JSON format"),
+        @ApiResponse(responseCode = "404", description = "The contingency list does not exists")})
+    public ResponseEntity<List<ContingencyInfos>> exportContingencyInfosList(@RequestParam(value = "networkUuid", required = false) UUID networkUuid,
```
Comment (antoinebhs) [inline src/main/java/org/gridsuite/actions/server/ContingencyListController.java:118]: we never return (responseCode = "404") from this endpoint, to remove ?

**Review by antoinebhs — CHANGES_REQUESTED** [review]
Same issue with sensitivity-analysis-server: https://github.com/gridsuite/sensitivity-analysis-server/blob/4350f48fac079443b9d3fb4551bc5920446b8110/src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilderService.java#L60-L62
Can we fix it in the mean time for the endpoint /contingency-lists/{id}/export (@SlimaneAmar)?

**Comment by SlimaneAmar** [comment]
> Same issue with sensitivity-analysis-server: https://github.com/gridsuite/sensitivity-analysis-server/blob/4350f48fac079443b9d3fb4551bc5920446b8110/src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilderService.java#L60-L62 Can we fix it in the mean time for the endpoint /contingency-lists/{id}/export (@SlimaneAmar)?

Yes

File: src/main/java/org/gridsuite/actions/server/ContingencyListController.java [hunk 2]
```diff
@@ -111,6 +111,16 @@ public ResponseEntity<List<ContingencyInfos>> exportContingencyInfosList(@PathVa
         return ResponseEntity.ok().body(service.exportContingencyInfosList(id, networkUuid, variantId));
     }
 
+    @GetMapping(value = "/contingency-lists/contingency-infos/export", produces = MediaType.APPLICATION_JSON_VALUE)
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/main/java/org/gridsuite/actions/server/ContingencyListController.java:114]: /contingency-lists/{id}/export endpoint is used in sensitivity-analysis-server indeed !
And it seems there is a lot to do in this micro-service as well !

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java [hunk 3]
```diff
@@ -173,6 +173,12 @@ public List<ContingencyInfos> exportContingencyInfosList(UUID id, UUID networkUu
         return evaluateContingencyList(findContingencyList(id, network), network);
     }
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/actions/server/ContingencyListService.java:175]: remove previous `exportContingencyInfosList` definition

File: src/main/java/org/gridsuite/actions/server/ContingencyListController.java [hunk 4]
```diff
@@ -111,6 +111,16 @@ public ResponseEntity<List<ContingencyInfos>> exportContingencyInfosList(@PathVa
...
+    @Operation(summary = "Evaluate and export a contingency infos list to JSON format")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The contingency list in JSON format"),
+        @ApiResponse(responseCode = "404", description = "The contingency list does not exists")})
+    public ResponseEntity<List<ContingencyInfos>> exportContingencyInfosList(@RequestParam(value = "networkUuid", required = false) UUID networkUuid,
+                                                                             @RequestParam(value = "variantId", required = false) String variantId,
+                                                                             @RequestParam(value = "ids", required = true) List<UUID> ids) {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/org/gridsuite/actions/server/ContingencyListController.java]: done

File: src/main/java/org/gridsuite/actions/server/ContingencyListController.java (hunk 1 above)
Comment (jamal-khey) [reply src/main/java/org/gridsuite/actions/server/ContingencyListController.java:108]: 404 can happen in 
```
ContingencyListService.java:179
.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Contingency list " + id + " not found"));
```

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java (hunk 3 above)
Comment (jamal-khey) [reply src/main/java/org/gridsuite/actions/server/ContingencyListService.java]: removed

File: src/main/java/org/gridsuite/actions/server/ContingencyListController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/org/gridsuite/actions/server/ContingencyListController.java:104]: will be done in a separate pull request

File: src/main/java/org/gridsuite/actions/server/ContingencyListController.java [hunk 5]
```diff
@@ -111,6 +111,16 @@ public ResponseEntity<List<ContingencyInfos>> exportContingencyInfosList(@PathVa
         return ResponseEntity.ok().body(service.exportContingencyInfosList(id, networkUuid, variantId));
     }
 
+    @GetMapping(value = "/contingency-lists/contingency-infos/export", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Evaluate and export a contingency infos list to JSON format")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The contingency list in JSON format"),
+        @ApiResponse(responseCode = "404", description = "The contingency list does not exists")})
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/main/java/org/gridsuite/actions/server/ContingencyListController.java]: fixed

**Review by sBouzols — APPROVED** [review]

**Pull request merged**